main.go

```diff
@@ -108,6 +111,19 @@ func setupConfig(c *cli.Context) {
 }
 
 func startNode(c *cli.Context) {
+	// Enable http profiling server if requested.
+	if cfg.Profile != "" {
+		go func() {
+			listenAddr := net.JoinHostPort("", cfg.Profile)
+			log.Infof("Profile server listening on %s", listenAddr)
+			profileRedirect := http.RedirectHandler("/debug/pprof",
+				http.StatusSeeOther)
+			http.Handle("/", profileRedirect)
+			ret := http.ListenAndServe(listenAddr, nil)
+			log.Errorf("Profile server ListenAndServe return %v", ret)
+		}()
+	}
+
```

Comment:
Please put the implementation in the util package and use the modularity functions.